packages/swing-store/src/transcriptStore.js

```diff
@@ -234,24 +237,51 @@ export function makeTranscriptStore(
     WHERE vatID = ? AND position < ?
   `);
 
-  /**
-   * End the current transcript span for a vat and start a new one.
-   *
-   * @param {string} vatID  The vat whose transcript is to rollover to a new
-   *    span.
-   */
-  function rolloverSpan(vatID) {
+  function doSpanRollover(vatID, isNewIncarnation) {
     ensureTxn();
-    const { hash, startPos, endPos } = getCurrentSpanBounds(vatID);
-    const rec = spanRec(vatID, startPos, endPos, hash, 0);
+    const { hash, startPos, endPos, incarnation } = getCurrentSpanBounds(vatID);
+    const rec = spanRec(vatID, startPos, endPos, hash, 0, incarnation);
     noteExport(spanMetadataKey(rec), JSON.stringify(rec));
     sqlEndCurrentSpan.run(vatID);
-    sqlWriteSpan.run(vatID, endPos, endPos, initialHash, 1);
-    const newRec = spanRec(vatID, endPos, endPos, initialHash, 1);
+    const incarnationToUse = isNewIncarnation ? incarnation + 1 : incarnation;
+    sqlWriteSpan.run(vatID, endPos, endPos, initialHash, 1, incarnationToUse);
+    const newRec = spanRec(
+      vatID,
+      endPos,
+      endPos,
+      initialHash,
+      1,
+      incarnationToUse,
+    );
     noteExport(spanMetadataKey(newRec), JSON.stringify(newRec));
     if (!keepTranscripts) {
       sqlDeleteOldItems.run(vatID, endPos);
     }
+    return incarnationToUse;
+  }
+
+  /**
+   * End the current transcript span for a vat and start a new one in a new incarnation.
```

Comment:
nit: add "(i.e. after a vat upgrade)" to the comment, to remind us what this is for